src/components/collection_vote_box/CollectionSubmitVoteBox.js

```diff
@@ -0,0 +1,141 @@
+import {div, h, span, textarea} from "react-hyperscript-helpers";
+import {useEffect, useState} from "react";
+import ld, {isNil} from "lodash";
+import {isEmpty} from "lodash/fp";
+import CollectionVoteYesButton from "./CollectionVoteYesButton";
+import CollectionVoteNoButton from "./CollectionVoteNoButton";
+import {Notifications} from "../../libs/utils";
+import {Votes} from "../../libs/ajax";
+
+const styles = {
+  baseStyle: {
+    fontFamily: 'Montserrat',
+    fontWeight: 'bold',
+    color: '#333F52',
+    display: 'flex',
+    flexDirection: 'column',
+    rowGap: '2rem'
+  },
+  question: {
+    fontSize: '1.8rem',
+  },
+  content: {
+    display: 'flex',
+    justifyContent: 'space-between'
+  },
+  subsection: {
+    display: 'flex',
+    flexDirection: 'column',
+    rowGap: '1.5rem'
+  },
+  voteButtons: {
+    display: 'flex',
+    columnGap: '1rem'
+  },
+  rationaleTextArea: {
+    borderRadius: '4px',
+    fontWeight: '500',
+    color: '#181818A6',
+    width: '40rem',
+  }
+};
+
+export default function CollectionSubmitVoteBox(props) {
+  const [vote, setVote] = useState();
+  const [rationale, setRationale] = useState('');
+  const [submitted, setSubmitted] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+  const {question, votes, isFinal, isLoading} = props;
+
+  useEffect(() => {
+    setDisabled(props.isDisabled || (isFinal && submitted) || isLoading);
+  }, [props.isDisabled, isFinal, submitted, isLoading])
+
+  useEffect(() => {
+    if (!isEmpty(votes)) {
+      const prevVote = votes[0];
+
+      const voteValues = ld.map(votes, vote => vote.vote);
+      if (allMatch(voteValues)) {
+        setVote(prevVote.vote);
+        setSubmitted(true);
+      }
+
+      const rationaleValues = ld.map(votes, vote => vote.rationale);
```

Comment:
When I tried switching some of my regular maps to the `lodash/fp` version, I started to experience weird bugs:

- ` map(vote => vote.vote)(votes)` led to all of the vote api calls to fail
- `map(vote => vote.rationale)(votes)` made the vote buttons and rationale textarea to be immediately disabled

I'm not sure why this would be the case because these bugs aren't directly related to the map calls